src/basilisp/contrib/nrepl_server.lpy

```diff
@@ -0,0 +1,478 @@
+        [v opts] v
+        ns (:ns opts)]
+    (swap! client* assoc :*1 v :*2 *1 :*3 *2)
+    (let [v (format-value (:nrepl.middleware.print/print request)
+                          (:nrepl.middleware.print/options request)
+                          v)]
+      (send-fn request {"value" (str v)
+                        "ns" (str ns)}))))
+
+(defn- handle-error [send-fn request e]
+  (let [{:keys [client* ns]} request
+        data (ex-data e)
+        message (or (:message data) (str e))]
+    (swap! client* assoc :*e e)
+    (send-fn request {"err" (str message)})
+    (send-fn request {"ex" (traceback/format_exc)
+                      "status" ["eval-error"]
+                      "ns" ns})))
+
+(defn- do-handle-eval
+  "Evaluate the ``request`` ``code`` of ``file`` in the ``ns`` namespace
+  according to the current state of the ``client*`` and sends its
+  result with ``send-fn``.
+
+  The result sent is either the last evaluated value or exception,
+  followed by the \"done\" status.
+
+  If ``ns`` is not provided, then it uses the ``client``'s :eval-ns as
+  the evaluation namespace. The latter is updated with the current
+  namespace after evaluation is completed.
+
+  It binds the `*1`, `*2`, `*3` and `*e` variables for evaluation from
+  the corresponding ones found in ``client*``, and updates the latter
+  according to the result."
+  [{:keys [client* code ns file _column _line] :as request} send-fn]
+  (let [{:keys [*1 *2 *3 *e eval-ns]} @client*]
+    (let [out-stream (StreamOutFn #(send-fn request {"out" %}))
+          reader (io/StringIO code)
+          ctx (basilisp.lang.compiler.CompilerContext. (or file "nREPL input"))
+          eval-ns (if ns
+                    (create-ns (symbol ns))
+                    eval-ns)]
+      (binding [*ns* eval-ns
+                *out* out-stream
+                *1 *1 *2 *2 *3 *3 *e *e]
+        (try
+          (let [results (for [form (seq (basilisp.lang.reader/read reader
+                                                                   *resolver*
+                                                                   *data-readers*))]
+                          (basilisp.lang.compiler/compile-and-exec-form form
+                                                                        ctx
+                                                                        *ns*))
+                result (last results)]
+            (send-value request send-fn [result {:ns (str *ns*)}]))
+          (catch python/Exception e
+            (info :eval-exception e)
+            (handle-error send-fn (assoc request :ns (str *ns*)) e))
+          (finally
+            (swap! client* assoc :eval-ns *ns*)
+            (send-fn request {"ns" (str *ns*)
+                              "status" ["done"]})))))))
+
+(defn- handle-eval [request send-fn]
+  (do-handle-eval request send-fn))
+
+(defn- handle-clone [request send-fn]
+  (send-fn request {"new-session" (str (uuid/uuid4))
```

Comment:
We have [`random-uuid`](https://github.com/basilisp-lang/basilisp/blob/ebfa81ba8a2b237242d044ec4f09b1835f2fb1fa/src/basilisp/core.lpy#L3351-L3354) now!

Reply:
How did I missed this? :) updated